src/rust/engine/fs/store/src/lib.rs

```diff
@@ -678,6 +678,7 @@ impl Store {
                   .await?;
                 Ok(Either::Right(reachable))
               }
+              Ok(Some(EntryType::Tree)) => unimplemented!(),
```

Comment:
@stuhood / @illicitonion: Does it make sense for trees to have reachability? Is it just the embedded directories?

Reply:
This method is used both for reachability, and for `ensure_remote_has_recursive` (although perhaps it never encounters a `Tree` in practice?). Reachability will use this method for any digest held in memory in `pantsd`, which will be `Snapshots`/`ProcessResults`, etc.

Depending on how you have implemented the conversion from `Tree` to `Directory`, it's possible that both `Snapshots` and `ProcessResults` end up holding directory-digests rather than tree-digests, and so a tree will never be reachable there. But Tree is a sortof interesting structure... currently to expand `Snapshots` we recursively expand `Directories` from the database: being able to load an entire Tree in one shot might be a useful performance boost.

If leaving this `unimplemented!` or `unreachable!` passes the tests, then it's likely safe by definition.

Reply:
`ensure_remote_has_recursive` won't encounter `Tree` at all because input roots are only ever specified as `Directory` protos. `Tree` is used on the way out as the result of remote execution requests and not on the way in.

Reply:
> Tree is used on the way out as the result of remote execution requests and not on the way in.

Makes sense. But `ProcessResult` deals with outputs, so hypothetically it could be adapted to hold a Tree-digest rather than a Directory-digest if we wanted it to. Ditto `Snapshot`, potentially. But as mentioned above, I think the primary reason to do that would be for performance, so if for the purposes of this patch we just convert from one to the other, that seems fine.